save-common/src/commonMain/kotlin/org/cqfn/save/core/files/FileUtils.kt

```diff
@@ -72,3 +72,35 @@ fun FileSystem.createFile(path: Path): Path {
 fun FileSystem.readLines(path: Path): List<String> = this.read(path) {
     generateSequence { readUtf8Line() }.toList()
 }
+
+/**
+ * Returns a sequence of underlying directories, filtering on every level by [directoryPredicate].
+ * Example:
+ * ```
+ * | directory1
+ * |_ file1
+ * | | directory11
+ * | |_ file2
+ * | |_ directory21
+ * | | directory12
+ * | | directory13
+ * | |_ directory23
+ * | | |_ file33
```

Comment:
```suggestion
 * |____directory1
 * |       |____file1
 * |       |____directory11
 * |               |____file2
 * |               |____directory21
 * |       |____directory12
 * |       |____directory13
 * |       |____directory23
 * |               |____file33
```

Reply:
I've changed it to more `tree`-like output, also added a test with the same example.